app/forms/incomeSources/add/BusinessTradeForm.scala

```diff
@@ -26,41 +26,29 @@ import scala.util.matching.Regex
 
 case class BusinessTradeForm(trade: String)
 
-object BusinessTradeForm {
+object BusinessTradeForm extends CustomConstraints {
 
-  private val validTrade: Regex = "^[A-Za-z0-9 ,.&'\\\\/-]+$".r
-
-  val maxLength = 35
+  val businessTrade = "business-trade"
+  val MIN_LENGTH = 2
+  val MAX_LENGTH = 35
+  val permittedChars: Regex = "^[A-Za-z0-9 ,.&'\\\\/-]+$".r
 
   val tradeEmptyError = "add-business-trade.form.error.empty"
   val tradeShortError = "add-business-trade.form.error.short"
   val tradeLongError = "add-business-trade.form.error.long"
   val tradeInvalidCharError = "add-business-trade.form.error.invalid"
   val tradeSameNameError = "You cannot enter the same trade and same business name"
```

Comment:
this should be in message file if used.

Reply:
It's unrelated to my code, your changes will fix this

Reply:
and yes I agree - it was a strange implementation